raiden/api/rest.py

```diff
@@ -1127,7 +1128,7 @@ def _withdraw(
                 channel_state.partner_state.address,
                 total_withdraw,
             )
-        except (NonexistingChannel, UnknownTokenAddress) as e:
+        except (NonexistingChannel, UnknownTokenAddress, ServiceRequestFailed) as e:
```

Comment:
I think it's good to handle this in the REST-api and not in the python-api!
This is because we have this behaviour also for direct-transfers, where the PFS is queried in a similar fashion.
Here we fail early instead of hoping that the transport still has a WebRTC channel open with that node.

I would only argue that the HTTP-code is maybe not the correct choice, since `400 - BAD REQUEST` indicates client error / malformed requests.

At this point we don't know what caused the ServiceRequestFailed, but it's all (most likely) an error on the PFS side.
**For consistency reasons (with failing payments) I'd rather suggest returning the CONFLICT status code,
although it's still debatable wether this is best suited.**

Reply:
I implemented above